Allows for a second migration source format.

brandonparsons opened the pull request:
My attempt to address #1894.

I'm not quite sure about the following line, as I haven't gotten into typespecs at all in Elixir yet:

`@spec run(Ecto.Repo.t, binary | [{integer, module}], atom, Keyword.t) :: [integer]`

josevalim commented on lib/ecto/migrator.ex:
```diff
@@ -124,7 +124,14 @@ defmodule Ecto.Migrator do
+  during the migration process. The other option is to pass a list of tuples
+  that identify the version number and migration modules to be run, for example:
+
+  `Ecto.Migrator.run(Repo, [{0, MyApp.Migration1}, {1, MyApp.Migration2}, ...], :up, opts)`
```
You can remove backticks and indent with four spaces.

brandonparsons replied:
✅

josevalim commented on lib/ecto/migrator.ex:
```diff
@@ -170,49 +177,55 @@ defmodule Ecto.Migrator do
   end
 
+  # This function will match specific version/modules passed into `Migrator.run`.
+  defp migrations_for(migration_source) do
```
You can add when `is_list(migration_source)` as a guard then.

brandonparsons replied:
✅

josevalim commented on lib/ecto/migrator.ex:
```diff
@@ -265,10 +273,28 @@ defmodule Ecto.Migrator do
   end
+  defp raise_no_migration_in_module(mod) do
+    raise Ecto.MigrationError,
+          "module #{mod} does not contain any Ecto.Migration"
```
"does not contain any Ecto.Migration" => "is not an Ecto.Migration"

josevalim replied:
`#{mod}` should be `#{inspect mod}` so we properly format the module name.

brandonparsons replied:
✅

brandonparsons commented:
Thanks for reviewing. Looks like I'm picking up Elixir reasonably well! Addressed your comments in the second commit. I can squash if it all looks good to you.

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
